templates/switchers.js

```diff
@@ -1,197 +1,194 @@
-        all_languages[current_language] = current_language;
-    }
-    buf.push('</select>');
-    return buf.join('');
+  for (const [language, title] of languages) {
+    const option = document.createElement('option');
+    option.value = language;
+    option.text = title;
+    if (language === _CURRENT_LANGUAGE) option.selected = true;
+    select.add(option);
   }
 
-  function navigate_to_first_existing(urls) {
-    // Navigate to the first existing URL in urls.
-    const url = urls.shift();
-    if (urls.length == 0 || url.startsWith("file:///")) {
-      window.location.href = url;
-      return;
-    }
+  return select;
+};
+
+/**
+ * Change the current page to the first existing URL in the list.
+ * @param {Array<string>} urls
+ * @private
+ */
+const _navigate_to_first_existing = (urls) => {
+  // Navigate to the first existing URL in urls.
+  for (const url of urls) {
     fetch(url)
-      .then(function(response) {
+      .then((response) => {
         if (response.ok) {
           window.location.href = url;
-        } else {
-          navigate_to_first_existing(urls);
         }
       })
-      .catch(function(error) {
-        navigate_to_first_existing(urls);
-      });
-  }
-
-  function on_version_switch() {
-    const selected_version = this.options[this.selectedIndex].value + '/';
-    const url = window.location.href;
-    const current_language = language_segment_from_url();
-    const current_version = version_segment_from_url();
-    const new_url = url.replace('/' + current_language + current_version,
-                                '/' + current_language + selected_version);
-    if (new_url != url) {
-      navigate_to_first_existing([
-        new_url,
-        url.replace('/' + current_language + current_version,
-                    '/' + selected_version),
-        '/' + current_language + selected_version,
-        '/' + selected_version,
-        '/'
-      ]);
-    }
+      .catch((err) => void err);
   }
 
-  function on_language_switch() {
-    let selected_language = this.options[this.selectedIndex].value + '/';
-    const url = window.location.href;
-    const current_language = language_segment_from_url();
-    const current_version = version_segment_from_url();
-    if (selected_language == 'en/') // Special 'default' case for English.
-      selected_language = '';
-    let new_url = url.replace('/' + current_language + current_version,
-                              '/' + selected_language + current_version);
-    if (new_url != url) {
-      navigate_to_first_existing([
-        new_url,
-        '/'
-      ]);
-    }
+  // if all else fails, redirect to the d.p.o root
+  window.location.href = '/';
+};
+
+/**
+ * Callback for the version switcher.
+ * @param {Event} event
+ * @returns {void}
+ * @private
+ */
+const _on_version_switch = (event) => {
+  if (_IS_LOCAL) return;
+
+  const selected_version = event.target.value;
+  // English has no language prefix.
```

Comment:
Not related to this PR, but I read through [PEP 545 – Python Documentation Translations](https://peps.python.org/pep-0545/) this weekend and it says:

> http://docs.python.org/en/ will redirect to http://docs.python.org/.

Currently `/en/` is 404, I wonder if we should add this redirect or if it was decided to be unnecessary?

Reply:
I've also seen that, though I never got around to updating the PEP. I think that we should remove it from the PEP as something that was never implemented, rather than adding another set of redirects to maintain.

Reply:
See https://github.com/python/peps/pull/4093